tools/gce-google-keys-to-cmek/main.py

```diff
@@ -0,0 +1,207 @@
+    parser.add_argument('--key-name', required=True, dest='keyName', action='store', type=str, help='Name of the key to encrypt the disks. Must be in the same zone as the instance.')
+    parser.add_argument('--key-version', required=True, dest='keyVersion', action='store', type=int, help='Version of the key to encrypt the disks.')
+    parser.add_argument('--destructive', dest='destructive', action='store_const', const=True, default=False, help='Upon completion, delete source disks and snapshots created during migration process.')
+    args = parser.parse_args()
+
+    migrate_instance_to_cmek(args.project, args.zone, args.instance, args.keyRing, args.keyName, args.keyVersion, args.destructive)
+
+def migrate_instance_to_cmek(project, zone, instance, keyRing, keyName, keyVersion, destructive):
+    start = time.time()
+
+    zone_regexp = r'^(\w\w-\w*\d)-(\w)$'
+    region = re.search(zone_regexp, zone).group(1)
+
+    compute = googleapiclient.discovery.build('compute', 'v1')
+
+    # TODO: Consider if we can use compute.disks().list() and do disk checking
+    # before we decide to stop the VM instance.
+    stop_instance(compute, project, zone, instance)
+    disks = get_instance_disks(compute, project, zone, instance)
+    for sourceDisk in disks:
+        disk_regexp = r'^https:\/\/www\.googleapis\.com\/compute\/v1\/projects\/(.*?)\/zones\/(.*?)\/disks\/(.*?)$'
+        disk_url = sourceDisk['source']
+        existing_disk_name = re.search(disk_regexp,disk_url).group(3)
+
+        if 'diskEncryptionKey' in sourceDisk:
+            print('Skipping {0}, already encrypyed with {1}', existing_disk_name, sourceDisk['diskEncryptionKey'])
+            continue
+
+        snapshot_name = existing_disk_name + '-goog-to-cmek'
+        new_disk_name = existing_disk_name + '-cmek'
+        disk_type = get_disk_type(compute, project, zone, existing_disk_name)
+
+        create_snapshot(compute, project, zone, existing_disk_name, snapshot_name)
+        key_name='projects/{0}/locations/{1}/keyRings/{2}/cryptoKeys/{3}/cryptoKeyVersions/{4}'.format(project, region, keyRing, keyName, keyVersion)
+        create_disk(compute, project, region, zone, snapshot_name, new_disk_name, disk_type, key_name)
+        detach_disk(compute, project, zone, instance, existing_disk_name)
+
+        boot = sourceDisk['boot']
+        autoDelete = sourceDisk['autoDelete']
+        attach_disk(compute, project, zone, instance, new_disk_name, boot, autoDelete)
+        if destructive:
+            delete_disk(compute, project, zone, existing_disk_name)
+            delete_snapshot(compute, project, snapshot_name)
+
+    start_instance(compute, project, zone, instance)
+
+    end = time.time()
+    print('Migration took {0} seconds.'.format(end -start))
+
+def get_disk_type(compute, project, zone, diskName):
```

Comment:
These functions all seem to take similar arguments
ie. `compute, project, zone`
Consider wrapping these in a simple class that takes these parameters once on instantiation and reading them from `self.<param>` in each individual method.